<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Henry Wittich" />

<meta name="date" content="2020-11-16" />

<title>relatedness-filtering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GWAS_QC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hwittich/GWAS_QC">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">relatedness-filtering</h1>
<h4 class="author">Henry Wittich</h4>
<h4 class="date">2020-11-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GWAS_QC/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201108code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201108)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201108code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201108)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhwittichGWASQCtree4f2f9baca44b262b543bceb684079cd9911e77c8targetblank4f2f9baa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/hwittich/GWAS_QC/tree/4f2f9baca44b262b543bceb684079cd9911e77c8" target="_blank">4f2f9ba</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomhwittichGWASQCtree4f2f9baca44b262b543bceb684079cd9911e77c8targetblank4f2f9baa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/hwittich/GWAS_QC/tree/4f2f9baca44b262b543bceb684079cd9911e77c8" target="_blank">4f2f9ba</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/relatedness-filtering.Rmd</code>) and HTML (<code>docs/relatedness-filtering.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/4f2f9baca44b262b543bceb684079cd9911e77c8/analysis/relatedness-filtering.Rmd" target="_blank">4f2f9ba</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
<td>
Completed PCA!
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hwittich/GWAS_QC/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/relatedness-filtering.html" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/72f56c0dadaccdf3edcce8a78d6628963577f392/analysis/relatedness-filtering.Rmd" target="_blank">72f56c0</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
<td>
Created IBD plots for each population group
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hwittich/GWAS_QC/af6a2e6a71991d9a849891b43edc2a5eaaae3106/docs/relatedness-filtering.html" target="_blank">af6a2e6</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/ad5a512aba99c8b56acaeac84f726f0d7579edad/analysis/relatedness-filtering.Rmd" target="_blank">ad5a512</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-17
</td>
<td>
Calculated IBD and generated an IBD plot for ALL population
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hwittich/GWAS_QC/42db0646b645aed6754113e117e6a1d2aa112ab9/docs/relatedness-filtering.html" target="_blank">42db064</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/af53890d0a90be014ddf15414a4f87d312f3a096/analysis/relatedness-filtering.Rmd" target="_blank">af53890</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-16
</td>
<td>
updated formatting on website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hwittich/GWAS_QC/55ee1d1efc872e3f892c9269cf53ad1bcdf74f5d/docs/relatedness-filtering.html" target="_blank">55ee1d1</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/9a7d97aa4ae2e1afb73a0606e0c2eb647405ed95/analysis/relatedness-filtering.Rmd" target="_blank">9a7d97a</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-16
</td>
<td>
Starting the relatedness filtering step
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The purpose of this step is to remove related individuals from the dataset. When running a GWAS, you want to have as little relatedness as possible because related individuals will have similar SNPs and thus could produce false positive significant signals.</p>
</div>
<div id="create-a-pruned-list-of-snp-ids" class="section level1">
<h1>1 Create a pruned list of SNP IDs</h1>
<p>Given the size of our dataset, calculating % identity by descent between every individual in the dataset using every SNP would be time-consuming and ultimately inefficient. According to Turner et al., a “dataset only 100,000 markers will yield stable estimates of kinship coefficients.” Furthermore, the IBD calculation assumes there is no LD. In this first step of relatedness filtering, we will generate a pruned set of SNPs that are in linkage equilibrium. This serves two purpose: first, it creates a smaller set up SNPs to calculate IBD with, and it also sets us up for LD pruning.</p>
<pre><code>plink --bed ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bed --bim ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bim --fam ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/11-16-2020/relatedness_steps/01LD_prune_list</code></pre>
<p>This command uses the HWE filtered list of SNPs and prunes them by linkage disequilibirum. the –indep-pairwise flag takes three parameters: the window size, the step size, and the r^2 threshold. In this case, plink looks at 50 variants at a time and calculates the squared correlation between pairs of variants; if a pair of variants has an r^2 higher than the threshold, they are pruned from the window. Plonk then shifts the window by 5 variants and repeats the calculations. In other words, for all SNPs in linkage disequilibrium with each other, all but one are pruned out of the dataset. SNPs that are in linkage equilibirum with each other are written to the prune.in file while those that are excluded are written to the prune.out file:<br />
– 01LD_prune_list.log<br />
– 01LD_prune_list.prune.in<br />
– 01LD_prune_list.prune.out</p>
</div>
<div id="determine-the-identity-by-descent-of-samples" class="section level1">
<h1>2 Determine the identity by descent of samples</h1>
<p>To gauge the number of related individuals in the dataset, we will be performing IBD calculations between every unique pair of individuals, using the pruned list of SNPs in linkage equilibrium. We can accomplish this with the plink –genome command:</p>
<pre><code>plink --bed ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bed --bim ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bim --fam ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.fam --extract ~/gwasqc/output/11-16-2020/relatedness_steps/01LD_prune_list.prune.in --genome --out ~/gwasqc/output/11-16-2020/relatedness_steps/02relatedness.genome</code></pre>
<p>Producing the following output files:<br />
– 02relatedness.genome.log<br />
– 02relatedness.genome.genome<br />
First, the –extract flag uses the list of SNP IDs generated in step one to pull all SNPs that are in linkage equilibirum out of the bed/bim/fam files. The –genome flag performs an IBD calculation for each unique pair of individuals in the dataset. This calculation produces the Z0, Z1, and Z2 estimates, which represent the proportion of loci where the two individuals share zero, one, or two alleles identical by descent. These values can be used to infer the relationship between the two individuals: <img src="assets/IBD_relationships.png" alt="Possible relationships" /></p>
<p>Thus, by plotting the Z0 versus Z1 for every pair of individuals, we get a good snapshot of the relationships in our sample population. This is called an IBD plot. Here is the IBD plot for all individuals in the dataset.</p>
<pre class="r"><code>library(&quot;ggplot2&quot;)
ibd &lt;- read.table(&quot;~/mount/gwasqc/output/11-16-2020/relatedness_steps/02relatedness.genome.genome&quot;,header=T)
ggplot(data=ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-1-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/af6a2e6a71991d9a849891b43edc2a5eaaae3106/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-1-1.png" target="_blank">af6a2e6</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Looking at the plot, it is clear that there are a few parent-child relationships in the dataset, as well as several siblings and cousins. Looking at the bottom right corner of the plot, there is a strange-looking gap in the cluster of non-related pairs. It is possible this is the result of the HapMap3 dataset being comprised of several populations from geographically distinct locations, resulting in strange relatedness estimates.</p>
</div>
<div id="ibd-separated-by-population" class="section level1">
<h1>3 IBD separated by population</h1>
<p>We are working with HapMap 3 data, which contains 1,301 samples from 11 different populations. The phase_3_samples.doc <a href="ftp://ftp.ncbi.nlm.nih.gov/hapmap/genotypes/hapmap3/plink_format/draft_2">stored with the HapMap data</a> lists the 11 populations and their corresponding 3-character labels, which are used in the relationships_w_pops_121708.txt file to identify samples by their population. For our purposes, we will group these 11 populations in 6 main groups:<br />
– African: ASW, LWK, MKK, and YRI<br />
– East Asian: CHB, CHD, and JPT<br />
– Northwestern European: CEU<br />
– Gujarati Indian: GIH<br />
– Mexican: MEX<br />
– Toscan: TSI</p>
<div id="extracting-sample-ids-by-population" class="section level2">
<h2>Extracting sample IDs by population</h2>
<p>Next, we must create a list of sample IDs for each of the 6 population groups that we can use to pull out their data and generate IBD plots. We can accomplish this with the UNIX grep command:</p>
<pre><code>#Pulling out all samples from the Gujarati Indian population
grep &quot;GIH&quot; relationships_w_pops_121708.txt &gt; ~/gwasqc/output/12-13-2020/separating_populations/gujarati_indian_IDs.txt</code></pre>
<p>The grep function takes 2 inputs: a string to search for and a file to search through. It searches through the file line by line and returns every line that contains the string. The “&gt;” character writes grep’s output to the file specified. With this command, I pulled out all of the samples in the dataset from the Gujarati Indian population and wrote their IDs to the file gujarati_indian_IDs.txt.</p>
<p>Let’s do the same for every other population group:</p>
<pre><code>#Pulling out all samples from African populations
#When searching for multiple terms, the syntax is as follows: &#39;word1\|word2&#39;
grep &#39;ASW\|LWK\|MKK\|YRI&#39; relationships_w_pops_121708.txt &gt; ~/gwasqc/output/12-13-2020/separating_populations/african_IDs.txt

#Pulling out all samples from East Asian populations
grep &#39;CHB\|CHD\|JPT&#39; relationships_w_pops_121708.txt &gt; ~/gwasqc/output/12-13-2020/separating_populations/east_asian_IDs.txt

#Pulling out all samples from the Northwestern European population
grep &quot;CEU&quot; relationships_w_pops_121708.txt &gt; ~/gwasqc/output/12-13-2020/separating_populations/northwestern_european_IDs.txt

#Pulling out all samples from the Mexican population
grep &quot;MEX&quot; relationships_w_pops_121708.txt &gt; ~/gwasqc/output/12-13-2020/separating_populations/mexican_IDs.txt

#Pulling out all samples from the Toscan population
grep &quot;TSI&quot; relationships_w_pops_121708.txt &gt; ~/gwasqc/output/12-13-2020/separating_populations/toscan_IDs.txt</code></pre>
</div>
<div id="creating-new-bfiles-by-population" class="section level2">
<h2>Creating new bfiles by population</h2>
<p>Using these lists of IDs, we can then extract the data for each sample and generate new bfiles, separated by population. Using the –keep and –make-bed plink commands, we can use our missingness- and hwe-filtered bfiles to make new bfiles that only keep individuals from the same population group.</p>
<pre><code>#Generating Gujarati Indian bfiles
plink --bfile ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE --keep ~/gwasqc/output/12-13-2020/separating_populations/gujarati_indian_IDs.txt --make-bed --out ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE

#Generating African bfiles
plink --bfile ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE --keep ~/gwasqc/output/12-13-2020/separating_populations/african_IDs.txt --make-bed --out ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE

#Generating East Asian bfiles
plink --bfile ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE --keep ~/gwasqc/output/12-13-2020/separating_populations/east_asian_IDs.txt --make-bed --out ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE

#Generating Northwestern European bfiles
plink --bfile ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE --keep ~/gwasqc/output/12-13-2020/separating_populations/northwestern_european_IDs.txt --make-bed --out ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE

#Generating Mexican bfiles
plink --bfile ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE --keep ~/gwasqc/output/12-13-2020/separating_populations/mexican_IDs.txt --make-bed --out ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE

#Generating Toscan bfiles
plink --bfile ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE --keep ~/gwasqc/output/12-13-2020/separating_populations/toscan_IDs.txt --make-bed --out ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE</code></pre>
</div>
<div id="ld-pruning-ibd-calculation-and-ibd-plots-by-population" class="section level2">
<h2>LD pruning, IBD calculation, and IBD plots by population</h2>
<div id="gujarati-indian" class="section level3">
<h3>Gujarati Indian</h3>
<pre><code>#LD Pruning
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/12-13-2020/relatedness_steps/01gujarati_indian_LD_prune_list

#IBD Calculation
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05gujarati_indian_filtered_HWE.fam --extract ~/gwasqc/output/12-13-2020/relatedness_steps/01gujarati_indian_LD_prune_list.prune.in --genome --out ~/gwasqc/output/12-13-2020/relatedness_steps/02gujarati_indian_relatedness</code></pre>
<pre class="r"><code>#Generating IBD Plot
gujarati_indian_ibd &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/02gujarati_indian_relatedness.genome&quot;,header=T)
ggplot(data=gujarati_indian_ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()+xlim(0,1)+ylim(0,1)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-2-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="african" class="section level3">
<h3>African</h3>
<pre><code>#LD Pruning
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/12-13-2020/relatedness_steps/01african_LD_prune_list

#IBD Calculation
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05african_filtered_HWE.fam --extract ~/gwasqc/output/12-13-2020/relatedness_steps/01african_LD_prune_list.prune.in --genome --out ~/gwasqc/output/12-13-2020/relatedness_steps/02african_relatedness</code></pre>
<pre class="r"><code>#Generating IBD Plot
african_ibd &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/02african_relatedness.genome&quot;,header=T)
ggplot(data=african_ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()+xlim(0,1)+ylim(0,1)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-3-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="east-asian" class="section level3">
<h3>East Asian</h3>
<pre><code>#LD Pruning
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/12-13-2020/relatedness_steps/01east_asian_LD_prune_list

#IBD Calculation
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05east_asian_filtered_HWE.fam --extract ~/gwasqc/output/12-13-2020/relatedness_steps/01east_asian_LD_prune_list.prune.in --genome --out ~/gwasqc/output/12-13-2020/relatedness_steps/02east_asian_relatedness</code></pre>
<pre class="r"><code>#Generating IBD Plot
east_asian_ibd &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/02east_asian_relatedness.genome&quot;,header=T)
ggplot(data=east_asian_ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()+xlim(0,1)+ylim(0,1)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-4-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="northwestern-european" class="section level3">
<h3>Northwestern European</h3>
<pre><code>#LD Pruning
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/12-13-2020/relatedness_steps/01northwestern_european_LD_prune_list

#IBD Calculation
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05northwestern_european_filtered_HWE.fam --extract ~/gwasqc/output/12-13-2020/relatedness_steps/01northwestern_european_LD_prune_list.prune.in --genome --out ~/gwasqc/output/12-13-2020/relatedness_steps/02northwestern_european_relatedness</code></pre>
<pre class="r"><code>#Generating IBD Plot
northwestern_european_ibd &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/02northwestern_european_relatedness.genome&quot;,header=T)
ggplot(data=northwestern_european_ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()+xlim(0,1)+ylim(0,1)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-5-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mexican" class="section level3">
<h3>Mexican</h3>
<pre><code>#LD Pruning
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/12-13-2020/relatedness_steps/01mexican_LD_prune_list

#IBD Calculation
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05mexican_filtered_HWE.fam --extract ~/gwasqc/output/12-13-2020/relatedness_steps/01mexican_LD_prune_list.prune.in --genome --out ~/gwasqc/output/12-13-2020/relatedness_steps/02mexican_relatedness</code></pre>
<pre class="r"><code>#Generating IBD Plot
mexican_ibd &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/02mexican_relatedness.genome&quot;,header=T)
ggplot(data=mexican_ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()+xlim(0,1)+ylim(0,1)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-6-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="toscan" class="section level3">
<h3>Toscan</h3>
<pre><code>#LD Pruning
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE.fam --indep-pairwise 50 5 0.3 --out ~/gwasqc/output/12-13-2020/relatedness_steps/01toscan_LD_prune_list

#IBD Calculation
plink --bed ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE.bed --bim ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE.bim --fam ~/gwasqc/output/12-13-2020/separating_populations/05toscan_filtered_HWE.fam --extract ~/gwasqc/output/12-13-2020/relatedness_steps/01toscan_LD_prune_list.prune.in --genome --out ~/gwasqc/output/12-13-2020/relatedness_steps/02toscan_relatedness</code></pre>
<pre class="r"><code>#Generating IBD Plot
toscan_ibd &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/02toscan_relatedness.genome&quot;,header=T)
ggplot(data=toscan_ibd,aes(x=Z0,y=Z1))+geom_point(alpha=1/4)+theme_bw()+xlim(0,1)+ylim(0,1)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hwittich/GWAS_QC/blob/7b0148c0ba32c566a6e7249090ef7c6eddb40538/docs/figure/relatedness-filtering.Rmd/unnamed-chunk-7-1.png" target="_blank">7b0148c</a>
</td>
<td>
Henry Wittich
</td>
<td>
2020-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="check-the-heterozygosity-distribution-of-samples-and-the-existence-of-outlier-samples" class="section level1">
<h1>4 Check the heterozygosity distribution of samples and the existence of outlier samples</h1>
<pre><code>plink --bed ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bed --bim ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bim --fam ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.fam --het --out ~/gwasqc/output/12-13-2020/relatedness_steps/03het_unfiltered</code></pre>
<p>The –het flag computes observed and expected autosomal homozygous genotype counts for each sample, stored in the O(HOM) and E(HOM) columns of the .het file, respectively. Expected number of homozygotes are determined using imputed minor allele frequencies.</p>
<pre class="r"><code>HET &lt;- read.table(&quot;~/mount/gwasqc/output/12-13-2020/relatedness_steps/03het_unfiltered.het&quot;,header=T,as.is=T)
H = (HET$N.NM. -HET$O.HOM.)/HET$N.NM.
oldpar=par(mfrow=c(1,2))
hist(H,50)
hist(HET$F,50)</code></pre>
<p><img src="figure/relatedness-filtering.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ld-prune-samples" class="section level1">
<h1>5 LD prune samples</h1>
<pre><code>plink --bed ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bed --bim ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.bim --fam ~/gwasqc/output/11-8-2020/missingness_hwe_steps/05filtered_HWE.fam --extract ~/gwasqc/output/11-16-2020/relatedness_steps/01LD_prune_list.prune.in --make-bed --out ~/gwasqc/output/12-13-2020/relatedness_steps/04LD_pruned</code></pre>
</div>
<div id="optionally-remove-duplicate-or-related-samples-and-regenerate-homozygosity-estimates-from-filtered-samples" class="section level1">
<h1>6 Optionally remove duplicate or related samples and regenerate homozygosity estimates from filtered samples</h1>
<p>For now, I am not doing this.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 19.10

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.8.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.8.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.2   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       pillar_1.4.6     compiler_3.6.1   later_1.1.0.1   
 [5] git2r_0.27.1     tools_3.6.1      digest_0.6.25    evaluate_0.14   
 [9] lifecycle_0.2.0  tibble_3.0.4     gtable_0.3.0     pkgconfig_2.0.3 
[13] rlang_0.4.7      rstudioapi_0.11  yaml_2.2.1       xfun_0.17       
[17] withr_2.3.0      stringr_1.4.0    knitr_1.29       fs_1.5.0        
[21] vctrs_0.3.5      rprojroot_1.3-2  grid_3.6.1       glue_1.4.2      
[25] R6_2.4.1         rmarkdown_2.3    farver_2.0.3     magrittr_1.5    
[29] whisker_0.4      backports_1.1.10 scales_1.1.1     promises_1.1.1  
[33] ellipsis_0.3.1   htmltools_0.5.0  colorspace_2.0-0 httpuv_1.5.4    
[37] labeling_0.4.2   stringi_1.5.3    munsell_0.5.0    crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
